Userland/Libraries/LibAudio/FlacLoader.cpp

```diff
@@ -181,8 +181,14 @@ ErrorOr<FlacRawMetadataBlock, LoaderError> FlacLoaderPlugin::next_meta_block(Big
     auto block_data_result = ByteBuffer::create_uninitialized(block_length);
     FLAC_VERIFY(!block_data_result.is_error(), LoaderError::Category::IO, "Out of memory");
     auto block_data = block_data_result.release_value();
-    // Reads exactly the bytes necessary into the Bytes container
-    LOADER_TRY(bit_input.read(block_data));
+
+    // Blocks might exceed the our buffer size.
+    auto bytes_left_to_read = block_data.bytes();
```

Comment:
Typo. Why are we even prevented from reading beyond our buffer size in the first place? The raw block data is large enough, after all...

Reply:
We are limited by the BufferedStream buffer size. https://github.com/SerenityOS/serenity/blob/417a385db143e57a36ad3179f1b3514e7c432730/Userland/Libraries/LibAudio/FlacLoader.cpp#L36
…if that’s what you meant.

Reply:
Yeah, but can't that internally just use its buffer multiple times, writing over to the output data if necessary?

Reply:
(I do realize you'll probably need to change Buffered, which is something I've done with varying success in the past)